### PR TITLE
Implement new module with EventBus API

### PR DESCRIPTION
**Pull request by JavaSaBr**
## Overview

This branch introduces a new `rlib-eventbus` module with a very typed low-overhead event bus API. It adds strongly typed subscription/publishing contracts, default implementation with sync/background dispatch, type-id consistency guards, comprehensive edge-case tests, API naming cleanup, and updated documentation (Javadocs, README usage examples, and event flow diagram).

### 1. New module: typed low-overhead EventBus
- Added a new Gradle module `:rlib-eventbus` and wired it into `settings.gradle`
- Added `rlib-eventbus/build.gradle` with publishing/java configuration and API dependency on `rlib-collections`
- Introduced API surface in `EventBus`:
  - `subscribe(...)` / `unsubscribe(...)`
  - `send(...)` (synchronous)
  - `sendInBackground(...)` (background dispatch)
  - Nested contracts `TypeIdSet`, `TypeIdFactory`, `TypeId`, and `Event`

### 2. Core implementation and dispatch model
- Added `DefaultEventBus` with:
  - Copy-on-modify consumer lists indexed by numeric `TypeId.id()`
  - Lazy type registration and sparse index expansion
  - Event-type to type-id consistency guard (`IllegalStateException` on conflicting IDs)
  - Validation for invalid negative type IDs (`IllegalArgumentException`)
  - Background dispatch through `ForkJoinPool.commonPool()`
- Added `DefaultTypeIdFactory` with per-`TypeIdSet` typed factories and stable per-event type IDs

### 3. Event flow diagram

```mermaid
flowchart TD
    A["Create TypeIdFactory<br/>EventBusFactory.createTypeIdFactory"] --> B["Create EventBus<br/>EventBusFactory.createEventBus"]
    B --> C["Resolve TypeId<br/>typeIdFactory.typeIdOf event type"]
    C --> D["Subscribe consumer<br/>eventBus.subscribe typeId consumer"]
    D --> E{Publish mode}
    E -->|send| F["Dispatch in caller thread"]
    E -->|sendInBackground| G["Schedule on executor"]
    G --> H["Dispatch on background thread"]
    F --> I["Lookup consumers by typeId id"]
    H --> I
    I --> J["Invoke subscribed consumers"]
```

### 4. Usage samples

Synchronous publish:

```java
interface AppEvents extends EventBus.TypeIdSet {}

record UserCreated(
    String userId,
    EventBus.TypeId<AppEvents, UserCreated> typeId
) implements EventBus.Event<AppEvents> {}

var typeIdFactory = EventBusFactory.createTypeIdFactory(AppEvents.class);
var eventBus = EventBusFactory.createEventBus(typeIdFactory);
var userCreatedTypeId = typeIdFactory.typeIdOf(UserCreated.class);

eventBus.subscribe(userCreatedTypeId, event ->
    System.out.println("Created: " + event.userId()));

eventBus.send(new UserCreated("user-42", userCreatedTypeId));
```

Background publish:

```java
eventBus.sendInBackground(new UserCreated("user-43", userCreatedTypeId));
```

### 5. Test coverage for happy paths and edge cases
- Added `DefaultEventBusTest` with **10 tests** covering:
  - Multi-consumer delivery across different event types
  - Negative `TypeId` handling for send/unsubscribe
  - Out-of-range/sparse `TypeId` behavior
  - Unsubscribe lifecycle (including duplicate unsubscribe)
  - Conflicting type-id detection for same event type
  - Background dispatch behavior with deterministic latch-based assertion

## Testing & Validation
- `./gradlew --no-daemon :rlib-eventbus:test` ✅
- `./gradlew --no-daemon :rlib-eventbus:compileJava` ✅
- Full branch verification before merge: `./gradlew clean build` (recommended)

### CODE REVIEW

File: rlib-eventbus/src/main/java/javasabr/rlib/eventbus/impl/DefaultEventBus.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  private synchronized void registerTypeId(TypeId<S, ? extends Event<S>> typeId) {
+    if (typeId.id() < 0) {
+      throw new IllegalArgumentException("Unexpected typeId:[%s]".formatted(typeId));
+    }
+    Class<? extends Event<S>> eventType = typeId.eventType();
+    TypeId<S, ? extends Event<S>> alreadyRegistered = knownEventTypes.get(eventType);
+    if (alreadyRegistered != null && alreadyRegistered != typeId) {
+      throw new IllegalStateException(
+          "EventType:[%s] is already registered with typeId:[%s], but now received typeId:[%s]".formatted(eventType, alreadyRegistered, typeId));
+    }
```
[the remark this answers is not in the document]
Comment (JavaSaBr) [reply rlib-eventbus/src/main/java/javasabr/rlib/eventbus/impl/DefaultEventBus.java:79]: it''s why we have typeIdFactory and typed generics to avoid any manual manipulations over this and some compile-time checks
